BatchedMesh: Add "toJSON" and "copy" support

gkjohnson opened the pull request:
Related issue: #22376, #27111 

**Description**

Add support for "toJSON" and "copy" to "BatchedMesh".

**Upcoming PRs**
- Raycasting, sorting support
- Object-level bounds
- Add support to ObjectLoader
- Move to core
- Documentation

gkjohnson committed:
Add a copy function

gkjohnson committed:
Add "toJSON" support

gkjohnson committed:
Remove toJSON override

gkjohnson committed:
copy paste error

Mugen87 commented on src/core/Object3D.js:
```diff
@@ -719,6 +719,26 @@ class Object3D extends EventDispatcher {
+			object._geometryInitialized = this._geometryInitialized;
+			object._geometryCount = this._geometryCount;
+
+			object._matricesTexture = this._matricesTexture.toJSON();
```
Can we avoid serializing the internal texture? Ideally, it is just recreated in `BatchedMesh` if it is missing.

Besides, can we avoid the underscore syntax in the JSON object? 

Also, at some point this logic should be moved to `BatchedMesh.toJSON()`. `Object3D` should access private variables of other subclasses.

gkjohnson replied:
> Can we avoid serializing the internal texture? Ideally, it is just recreated in BatchedMesh if it is missing.

This texture is how the matrices are canonically stored. I can generate a list of serialized matrices from the texture instead, though, if that's acceptable.

> Besides, can we avoid the underscore syntax in the JSON object?

Yup! I'll fix it.

> Also, at some point this logic should be moved to BatchedMesh.toJSON(). Object3D should not access private variables of subclasses.

I thought so, too, but this looked like how all the other classes like InstancedMesh were working. Though it's unclear to me why this is needed.

Mugen87 replied:
> This texture is how the matrices are canonically stored. I can generate a list of serialized matrices from the texture instead, though, if that's acceptable.

I forgot after #27130 there is only the texture that holds the matrices. In this case, I'm fine with serializing directly. the texture

Mugen87 replied:
BTW: `ObjectLoader` should also be able to handle `BatchedMesh`. Maybe we can add this to the todo list.

Mugen87 commented:
Sorry, approved a bit too early. The removal of the underscore in the JSON would still be nice^^!

gkjohnson committed:
Remove underscores from json

gkjohnson commented:
> Sorry, approved a bit too early. The removal of the underscore in the JSON would still be nice^^!

Fixed!

> BTW: ObjectLoader should also be able to handle BatchedMesh. Maybe we can add this to the todo list.

Added to the list!

Methuselah96 commented on examples/jsm/objects/BatchedMesh.js:
```diff
@@ -563,17 +563,29 @@ class BatchedMesh extends Mesh {
 	}
 
-	copy() {
+	copy( source ) {
```
Can we have `copy` return `this` to make it consistent with parent classes `Object3D` and `Mesh`?
